pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/RangeIndexReaderImpl.java

```diff
@@ -217,4 +272,26 @@ public void close() {
     // NOTE: DO NOT close the PinotDataBuffer here because it is tracked by the caller and might be reused later. The
     // caller is responsible of closing the PinotDataBuffer.
   }
+
+  private ImmutableRoaringBitmap getMatchesInRange(int firstRangeId, int lastRangeId) {
+    if (firstRangeId == lastRangeId) {
+      return null;
+    }
+    MutableRoaringBitmap matching = new MutableRoaringBitmap();
+    for (int rangeId = firstRangeId + 1; rangeId < lastRangeId; ++rangeId) {
```

Comment:
Won't work if `lastRangeId` is -1?

Reply:
(nit) We usually put `rangeId++`

Reply:
No, this is correct. `firstRangeId == 0` means that the lower bound was below the lowest bucket, so range 0 (`firstRangeId + 1`) is within the range of the query so long as `lastRangeId > 0`. Similarly, a `lastRangeId == -1` means that the upper bound of the query was below the first range, which means that none of the rows are within the range of the query, so an empty bitmap is produced.

Reply:
Based on my reading, the `findRangeId()` will return `-1` if the value is smaller than the lower bound of the lowest bucket or larger than the higher bound of the highest bucket. What if the `lastRangeId == -1` because it is larger than the higher bound?

Reply:
Actually, Looking back at `findRangeId`, it is ambiguous. -1 can mean smaller than the lowest range or greater than the highest range. So this was always broken.

This code was lifted straight from `RangeIndexBasedFilterOperator`:
```java
// Ranges in the middle of first and last range are fully matched
     for (int rangeId = firstRangeId + 1; rangeId < lastRangeId; rangeId++) {
       docIds.or(rangeIndexReader.getDocIds(rangeId));
     }
```

I don't want existing bugs/specification flaws to get in the way of redefining this functionality.